README.md

```diff
@@ -14,134 +20,432 @@ For suggestions on improvements,
+user-defined transformations. The transformer is then applied to the training dataset and the validation dataset, 
+creating transformed datasets that are used by subsequent steps for estimator training and model performance evaluation.
+
+The user-defined transformation function should be written in [`steps/transform.py`](https://github.com/mlflow/mlp-regression-template/blob/main/steps/transform.py), 
+and should return an unfitted estimator that is sklearn-compatible; that is, the returned object should define 
+`fit()` and `transform()` methods. `steps/transform.py` contains an example placeholder function.
+
+The transform step is configured by the `steps.transform` section in pipeline.yaml:
+<details>
+<summary>Reference</summary>
+
+- `transformer_method`: string. Optional.  
+Fully qualified name of the method that returns an `sklearn`-compatible transformer which applies feature 
+transformation during model training and inference. If absent, the identity transformer will be used.  
+<u>Example</u>:
+  ```
+  transformer_method: steps.split.transformer_fn
+  ```
+
+</details>
+
+**Step artifacts**:
+- `transformed_training_data`: transformed training dataset as a Pandas DataFrame.
+- `transformed_validation_data`: transformed validation dataset as a Pandas DataFrame.
+- `transformer`: the sklearn transformer.
+
+
+### Train step
+The train step uses the transformed training dataset output from the transform step to fit an user-defined estimator. 
+The estimator is then joined with the fitted transformer output from the transform step to create a model pipeline. 
+Finally, this model pipeline is evaluated against the transformed training and validation datasets to compute performance metrics.  
+
+Custom evaluation metrics are computed according to definitions in [`steps/custom_metrics.py`](https://github.com/mlflow/mlp-regression-template/blob/main/steps/custom_metrics.py)
+and the `metrics` section of `pipeline.yaml`; see [Custom Metrics](#custom-metrics) section for reference. 
+
+The model pipeline and its associated parameters, performance metrics, and lineage information are logged to [MLflow Tracking](https://www.mlflow.org/docs/latest/tracking.html), producing an MLflow Run.
+
+The user-defined estimator function should be written in [`steps/train.py`](https://github.com/mlflow/mlp-regression-template/blob/main/steps/train.py), 
+and should return an unfitted estimator that is `sklearn`-compatible; that is, the returned object should define 
+`fit()` and `transform()` methods. `steps/train.py` contains an example placeholder function.
+
+The train step is configured by the `steps.train` section in pipeline.yaml:
+<details>
+<summary>Reference</summary>
+
+- `estimator_method`: string. Required.  
+Fully qualified name of the method that returns an `sklearn`-compatible estimator used for model training.  
+<u>Example</u>:
+  ```
+  estimator_method: steps.train.estimator_fn
+  ```
+</details>
```

Comment:
should we add a note about the `using: estimator_spec` plugin?

Reply:
We left that out because it's the only option right now. We'll revisit it when autoML support is included